### DISCUSSION TITLE
Select relation field problem

### DISCUSSION THREAD

**Discussion by ifrvn** [prisma/prisma#20185] [Q&A]
I'm using prisma 5.0.0 with nestjs 10.0.0
I have a User model with some relation fields:

```prisma
model User {
  id               Int          @id @default(autoincrement())
  authingId        String       @unique @db.VarChar(32)
  email            String?      @unique
  emailVerified    Boolean      @default(false)
  phone            String?      @unique
  phoneCountryCode String?
  phoneVerified    Boolean      @default(false)
  username         String       @unique
  photo            String?
  workStatus       WorkStatus   @default(Active)
  roles            Role[]
  createdAt        DateTime     @default(now())
  updatedAt        DateTime     @updatedAt
  lastLogin        DateTime?
  lastIp           String?
  lastSync         DateTime?
  profile          Profile?
  followers        Follow[]     @relation("following")
  following        Follow[]     @relation("follower")
  posts            Post[]
  articles         Article[]
  videos           Video[]
  comments         Comment[]
  webaMember       WebaMember[] @relation("user")

  @@index([email, phone, username])
}
```

In UserService, I have this `user` function to select a user. and I want to use the `include` param to let caller decide whether/which relation fields are needed.
```typescript
import { Injectable, Logger } from '@nestjs/common'
import { ConfigService } from '@nestjs/config'
import { Prisma } from '@prisma/client'
import { PrismaService } from '../common/prisma.service'

@Injectable()
export class UsersService {
  constructor(
    private readonly prisma: PrismaService
  ) {}

  async user(
    where: Prisma.UserWhereUniqueInput,
    include?: Prisma.UserInclude
  ) {
    const user = await this.prisma.user.findUnique({
      where,
      include,
    })
    return user
  }
 
 //...
 }
 ```

But when I call it in `controller`, ts error me that the relation field does not exist on the type of user even though I've passed the include param.

```typescript
  const user = await this.usersService.user(
    { id: req.user.id },
    { followers: true }
  )
  if (user?.followers) {
    return
  }
```

![image](https://github.com/prisma/prisma/assets/21982381/33c09068-d2de-494c-92d6-49ae7f6706fc)

how to fix this issue, or is there any better solution for this?

**Comment by ludralph** [comment]
Hi @ifrvn 👋 

Thank you for raising this question. Can you please try and check if it works for you in Prisma version 4.15.0. This issue should be potentially fixed in Prisma version 5.1.0 as `GetResult` type would be going away.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by ifrvn** [reply]
Thank you! It works in Prisma version 4.15.0

**Reply by ifrvn** [reply]
I have no idea but after I restarted the computer, it doesn't work again 😭

// In controller
![image](https://github.com/prisma/prisma/assets/21982381/a3193a65-46c0-4aff-93b0-a1581ca5c77f)
![image](https://github.com/prisma/prisma/assets/21982381/28272230-dc18-4ebf-b9f2-4096967052ab)

// UsersService.user
![image](https://github.com/prisma/prisma/assets/21982381/ad9d08d8-5640-4bdb-bdb0-ef5940af5d7e)

prisma and @prisma/client are both set to version "4.15.0"

**Reply by ludralph** [reply]
@ifrvn, Can you please share a repository so I can take a look?

**Reply by ifrvn** [reply]
Thank you for your reply! It turns out that the reason is because I had set `strictNullChecks` to true in my tsconfig. After changing it to false, everything worked fine.

**Reply by nurul3101** [reply]
Thanks for letting us know your solution 🙌

**Reply by ifrvn** [reply]
@ludralph I just realized that this might not be a proper solution. It would just simply gives me all the relation fields instead of the included ones. So I set `strictNullChecks` back to false and created a [repo](https://github.com/ifrvn/nestjs-prisma-test)

You should see the error at 
https://github.com/ifrvn/nestjs-prisma-test/blob/ba42a5546cc6f1db9ea49828e17b95f44fdb2a19/src/app.service.ts#L19C22-L19C22

**Reply by mlbiche** [reply]
Is there any move on this issue ? Is it really fixed ?
